app/controllers/DraftController.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions.Actions
+import models.domain.UserAnswersReader
+import models.journeyDomain.PreTaskListDomain
+import models.{LocalReferenceNumber, NormalMode}
+import navigation.PreTaskListNavigatorProvider
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+
+import javax.inject.Inject
+
+class DraftController @Inject() (
+  val controllerComponents: MessagesControllerComponents,
+  actions: Actions,
+  navigatorProvider: PreTaskListNavigatorProvider
+) extends FrontendBaseController {
+
+  def draftRedirect(lrn: LocalReferenceNumber): Action[AnyContent] = actions.requireData(lrn) {
+    implicit request =>
+      UserAnswersReader[PreTaskListDomain].run(request.userAnswers) match {
+        case Left(value) =>
+          Redirect(navigatorProvider(NormalMode).nextPage(request.userAnswers))
+        case Right(value) =>
+          Redirect(controllers.routes.TaskListController.onPageLoad(lrn))
+      }
```

Comment:
I believe the task list controller action builder essentially does this check anyway so we may be able to get away with just redirecting to the task list controller

Reply:
@jackheslop96 Looking at that. If they have completed part, but not all, of the pre-task-list it would always take them back to the LRN page whereas the way we have done it would take them to wherever they were up to in the pre-task-list. We don't want them to start again if they have not completed the pre-task-list we would just want to take them to the page that they are up to.